src/wwwroot/js/genpage/helpers/generatehandler.js

```diff
@@ -199,6 +199,10 @@ class GenerateHandler {
                         }
                         this.setImageFor(imgHolder, data.gen_progress.preview);
                     }
+                    if (data.gen_progress.preview && imageFullView.isOpen() && imageFullView.imgElement && imageFullView.currentBatchId == thisBatchId)
```

Comment:
this block should be inside the block above

Reply:
I would disagree here.  It's a check that exclusively applies to the Full View.  The block above is a check that only applies to the Current Image.  The only part that really applies to all three is the `data.gen_progress.preview` part.

Reply:
That is... not correct.